network-mux/network-mux.cabal

```diff
@@ -112,6 +112,30 @@ library
     -Wredundant-constraints
     -Wunused-packages
 
+library traces
+  build-depends:
+    aeson,
+    base >=4.14 && <4.22,
+    formatting,
+    network-mux,
+    trace-dispatcher ^>= 2.10.0
+  hs-source-dirs: traces
+  visibility: public
+  exposed-modules:
+    Network.Mux.Traces
```

Comment:
We already have a very similar module `Network.Mux.Trace`, what about `Network.Mux.Trace.Instances`?